### ISSUE TITLE
Add default configuration

### ISSUE THREAD

**Issue by m1guelpf** [svenluijten/artisan-view#28]
I want to apply some options to all the views I create unless I say the opposite, like extending a certain layout or having certain elements. This could be archieved by adding a config file.

**Comment by svenluijten** [comment]
I feel like this would add unnecessary bloat / complications to a simple `php artisan make:*` package. How would the options work if this was implemented? Add new ones like `--no-sections`? Override them with empty values (`--section=`)?

**Comment by m1guelpf** [comment]
It would take the values from the config if defined, then check if there are options and overwrite if they are.

**Comment by svenluijten** [comment]
Right, so lets say I have a config file set up like this (mockup, obviously): 

```php
return [
    'extend' => 'layouts.app',

    'sections' => [
        'title',
        'content',
    ],
];
```

This would then always generate a view like this:

```blade
@extends('layouts.app')

@section('title')

@endsection

@section('content')

@endsection
```

What if I wanted to create a view that doesn't include the `@extends()` and `@section()`s? At the moment I can think of a few ways this could work, some maybe better than others 😉: 

```bash
$ php artisan make:view my-view --section=<X> --extend=<X> # where <X> is a predefined keyword.
$ php artisan make:view my-view --no-config
$ php artisan make:view my-view # and only read the config file if a --with-defaults flag is set, maybe?
```

**Comment by m1guelpf** [comment]
I'd say, if you have a config, you have to overwrite it with options. If you find yourself overwriting it very often, then you migth want to remove that config option...

**Comment by svenluijten** [comment]
Hmm. I'm very much in two minds about this. I don't see myself (or many other people) reach for this. I mean, how often do you create a view that you would need defaults?

On the other hand, it could be pretty cool to have for the power-user, and you could even take it a step further and introduce the concept of presets: 

```php
return [
    'presets' => [
        'foo' => [
            'extends' => 'layouts.foo',
            'sections' => ['title', 'content'],
        ],

        //
    ],
];
```

```sh
$ php artisan make:view --preset=foo
```

**Comment by m1guelpf** [comment]
I normally extend layouts.app always except on the landing page, and I'm sure most applications with a dashboard do this. Also, I love the preset idea!

**Comment by chrisdicarlo** [comment]
I like the preset idea.  Having them in a config would allow multiple presets, if you had for example a couple of standard layouts spread across the app.

I kind of think there should not be a default though - force the user to explicitly pass the preset on the command-line.

**Comment by svenluijten** [comment]
Yes, absolutely. I'm not a fan of adding a seperate config file just for this package though (I want it tightly integrated/coupled to Laravel), so what do you think of adding a `presets` key to just the standard `view` configuration file?

**Comment by chrisdicarlo** [comment]
I like that idea.  I agree a separate config just to hold preset values seems like overkill at the moment.  Would need to see if there are any other packages doing the same to make sure we don't have a name collision on the property but that should be pretty easy to find and even if a collision occurs it's a simple workaround.

Could even add a .env value to point to the valid key and let the user deal with namespace collisions if they happen to have another package using the same key.

**Comment by svenluijten** [comment]
Well, the results to my [totally scientific poll](https://twitter.com/svenluijten/status/865993973252403200) lead me to believe not a lot of people edit the config file anyway, and if they do, they can file an issue and we'll deal with that once we cross that bridge. YAGNI :smile:

**Comment by chrisdicarlo** [comment]
Agreed! 😆

**Comment by svenluijten** [comment]
This package has been abandoned because of the introduction of the new `make:view` command included in Laravel since [v10.23.0](https://github.com/laravel/framework/releases/tag/v10.23.0). For that reason, this issue will _not_ be worked on.
